evm_arithmetization/src/prover.rs

```diff
@@ -170,6 +202,427 @@ where
+        D,
+    >(
+        all_stark.keccak_sponge_stark,
+        config,
+        &trace_poly_values[*Table::KeccakSponge],
+        &ctl_data_per_table[*Table::KeccakSponge],
+        ctl_challenges,
+    ));
+
+    // Logic.
+    res.push(auxiliary_columns_single_stark::<F, C, LogicStark<F, D>, D>(
+        all_stark.logic_stark,
+        config,
+        &trace_poly_values[*Table::Logic],
+        &ctl_data_per_table[*Table::Logic],
+        ctl_challenges,
+    ));
+
+    // Memory.
+    res.push(
+        auxiliary_columns_single_stark::<F, C, MemoryStark<F, D>, D>(
+            all_stark.memory_stark,
+            config,
+            &trace_poly_values[*Table::Memory],
+            &ctl_data_per_table[*Table::Memory],
+            ctl_challenges,
+        ),
+    );
+
+    res
+}
+
+fn auxiliary_columns_single_stark<F, C, S, const D: usize>(
+    stark: S,
+    config: &StarkConfig,
+    trace_poly_values: &[PolynomialValues<F>],
+    ctl_data: &CtlData<F>,
+    ctl_challenges: &GrandProductChallengeSet<F>,
+) -> Vec<PolynomialValues<F>>
+where
+    F: RichField + Extendable<D>,
+    C: GenericConfig<D, F = F>,
+    S: Stark<F, D>,
+{
+    let rate_bits = config.fri_config.rate_bits;
+    let constraint_degree = stark.constraint_degree();
+    assert!(
+        constraint_degree <= (1 << rate_bits) + 1,
+        "The degree of the Stark constraints must be <= blowup_factor + 1"
+    );
+
+    let lookup_challenges: Vec<_> = ctl_challenges.challenges.iter().map(|ch| ch.beta).collect();
+    // Add lookup columns.
+    let lookups = stark.lookups();
+    let mut res = {
+        let mut columns = Vec::new();
+        for lookup in &lookups {
+            for &challenge in lookup_challenges.iter() {
+                columns.extend(lookup_helper_columns(
+                    lookup,
+                    trace_poly_values,
+                    challenge,
+                    constraint_degree,
+                ));
+            }
+        }
+        columns
+    };
+    let num_lookup_columns = res.len();
+
+    // Add CTL columns.
+    if let Some(p) = get_ctl_auxiliary_polys(Some(ctl_data)) {
+        res.extend(p);
+    }
+
+    debug_assert!(
+        (stark.uses_lookups() || stark.requires_ctls()) || get_ctl_auxiliary_polys(Some(ctl_data)).is_none(),
+        "There should be auxiliary polynomials if and only if we have either lookups or require cross-table lookups."
+    );
+
+    res
+}
+
+/// Generates all quotient polynomials.
+fn all_quotient_polys<F, C, const D: usize>(
+    all_stark: &AllStark<F, D>,
+    trace_commitment: &BatchFriOracle<F, C, D>,
+    auxiliary_commitment: &BatchFriOracle<F, C, D>,
+    lookup_challenges: Option<Vec<F>>,
+    ctl_data_per_table: &[CtlData<F>; NUM_TABLES],
+    alphas: Vec<F>,
+    config: &StarkConfig,
+) -> Vec<Vec<PolynomialValues<F>>>
+where
+    F: RichField + Extendable<D>,
+    C: GenericConfig<D, F = F>,
+{
+    let res = Vec::new();
+
+    // TODO: Implement it.
```

Comment:
Ideally we would want to use the existing `compute_quotient_polys` Plonky2 method, but it takes `PolynomialBatch` as input instead of `BatchFriOracle`.